lib/User/Backend.php

```diff
@@ -252,45 +252,52 @@ public function getCurrentUserId() {
 				// find user id through different token validation methods
 				foreach ($this->tokenValidators as $validatorClass) {
 					$validator = \OC::$server->get($validatorClass);
-					$userId = $validator->isValidBearerToken($provider, $headerToken);
-					if ($userId) {
+					$sub = $validator->isValidBearerToken($provider, $headerToken);
+					if ($sub) {
```

Comment:
I agree `$userId` was not appropriated here but `$sub` is confusing as it's an attribute of the ID token but we might use/map another one.

How about `$oidcUserId` or `$tokenUserId`?

Same remark for all appearances of `$sub` :grin:.

Reply:
I renamed it to `tokenUserId`